File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/common/properties/JwtProperties.java
package com.multicampus.gamesungcoding.a11ymarketserver.common.properties;

import io.jsonwebtoken.io.Decoders;
import io.jsonwebtoken.security.Keys;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import javax.crypto.SecretKey;

@Component
@ConfigurationProperties(prefix = "jwt")
public class JwtProperties {

    @Getter
    private String secret;

    @Setter
    @Getter
    private long accessTokenValidityMs;

    @Setter
    @Getter
    private long refreshTokenValidityMs;

    @Getter
    private SecretKey secretKey;

    public void setSecret(String secret) {
        byte[] keyBytes = Decoders.BASE64.decode(secret);
        this.secretKey = Keys.hmacShaKeyFor(keyBytes);
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/common/properties/JwtProperties.kt
package com.multicampus.gamesungcoding.a11ymarketserver.common.properties

import io.jsonwebtoken.io.Decoders
import io.jsonwebtoken.security.Keys
import org.springframework.boot.context.properties.ConfigurationProperties
import javax.crypto.SecretKey

@ConfigurationProperties(prefix = "jwt")
data class JwtProperties(
    val secret: String,
    val accessTokenValidityMs: Long,
    val refreshTokenValidityMs: Long
) {
    val secretKey: SecretKey by lazy {
        val keyBytes = Decoders.BASE64.decode(secret)
        Keys.hmacShaKeyFor(keyBytes)
    }
}
